crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -239,7 +239,25 @@ impl SettlementEncoder {
             })
     }
 
-    pub fn finish(self) -> EncodedSettlement {
+    fn drop_unnecessary_tokens_and_prices(&mut self) {
+        let traded_tokens: HashSet<_> = self
+            .trades()
+            .iter()
+            .flat_map(|trade| {
+                [
+                    trade.order.order_creation.buy_token,
+                    trade.order.order_creation.sell_token,
+                ]
+            })
+            .collect();
+
+        self.tokens.retain(|token| traded_tokens.contains(token));
```

Comment:
I know this happens in finish anyway, but it would be clearer to also retain on self.clearing_prices here? Considering the function is called "and prices" .

Reply:
Done 👍